packages/teleport/src/DesktopSession/useDesktopSession.tsx

```diff
@@ -42,14 +43,55 @@ export default function useDesktopSession(ctx: Ctx) {
 
   const { username, desktopName, clusterId } = useParams<UrlDesktopParams>();
   const [hostname, setHostname] = useState<string>('');
+
+  const isUsingChrome = navigator.userAgent.indexOf('Chrome') > -1;
+  const clipboardEnabled = ctx.storeUser.getClipboardAccess();
+  const clipboardRWPermission = useClipboardRW(clipboardEnabled, 'prompt');
+  const [clipboard, setClipboard] = useState({
+    enabled: clipboardEnabled,
+    permission: clipboardRWPermission,
+    hasError: false,
+    errorText: '',
+  });
+
   const clientCanvasProps = useTdpClientCanvas({
     username,
     desktopName,
     clusterId,
     setTdpConnection,
     setWsConnection,
+    clipboardEnabled,
   });
 
+  useEffect(() => {
+    // errors:
+    // - role permits, browser not chromium
+    // - role permits, clipboard permissions denied
+    if (clipboardEnabled && !isUsingChrome) {
+      setClipboard({
+        enabled: clipboardEnabled,
+        permission: clipboardRWPermission,
+        hasError: true,
+        errorText:
+          'Your user role supports clipboard sharing over desktop access, however this feature is only available on chromium based browsers like Brave or Google Chrome. Please switch to a supported browser.',
```

Comment:
```suggestion
          'Your user role supports clipboard sharing over desktop access, however this feature is only available on Chromium based browsers like Brave or Google Chrome. Please switch to a supported browser.',
```

I'm not sure the average user will know what a "Chromium-based" browser is, and whether we should list examples of browsers that are Chromium but not Chrome.

@xinding33 or @ptgott - do either of you have suggestions for the copy here?

Reply:
Should we say something like this, providing a quick list of browsers that work?

```
Your user role supports clipboard sharing over desktop access. However this feature is only available in the following browsers:
- Microsoft Edge (after January 2020)
- Google Chrome
- Brave
- Opera
- Other browsers based on Chromium
Please switch to a supported browser.
```
I'm not sure how popular Brave, Opera, etc. are for our user base, so we can swap these out as needed. I'm also not sure if this unordered will fit in the UI. What do you think, @xinding33?

Reply:
The original copy was just me trying to push my pro-Brave agenda in an error message. Probably `"Your user role supports clipboard sharing over Teleport Desktop Access, however this feature is only available on Chromium based browsers like Google Chrome. Please switch to a supported browser."` would suffice and be understood by all. (But I'm still partial towards including Brave propaganda).